lib/python/picongpu/pypicongpu/laser.py

```diff
@@ -102,3 +108,80 @@ def _get_serialized(self) -> dict:
                 },
             },
         }
+
+
+@typeguard.typechecked
+class PlaneWaveLaser(Laser):
+    """
+    PIConGPU Plane Wave Laser
+
+    Holds Parameters to specify a plane wave laser
+    """
+
+    _name = "planewave"
+
+    class PolarizationType(enum.Enum):
+        """represents a polarization of a laser (for PIConGPU)"""
+
+        LINEAR = 1
+        CIRCULAR = 2
+
+        def get_cpp_str(self) -> str:
+            """retrieve name as used in c++ param files"""
+            cpp_by_ptype = {
+                PlaneWaveLaser.PolarizationType.LINEAR: "Linear",
+                PlaneWaveLaser.PolarizationType.CIRCULAR: "Circular",
+            }
+            return cpp_by_ptype[self]
+
+    wavelength = util.build_typesafe_property(float)
+    """wave length in m"""
+    duration = util.build_typesafe_property(float)
+    """duration in s (1 sigma)"""
+    focus_pos = util.build_typesafe_property(typing.List[float])
+    """focus position vector in m"""
+    phase = util.build_typesafe_property(float)
+    """phase in rad, periodic in 2*pi"""
+    E0 = util.build_typesafe_property(float)
+    """E0 in V/m"""
+    pulse_init = util.build_typesafe_property(float)
+    """laser will be initialized pulse_init times of duration (unitless)"""
+    propagation_direction = util.build_typesafe_property(typing.List[float])
+    """propagation direction(normalized vector)"""
+    polarization_type = util.build_typesafe_property(PolarizationType)
+    """laser polarization"""
+    polarization_direction = util.build_typesafe_property(typing.List[float])
+    """direction of polarization(normalized vector)"""
+    laser_nofocus_constant_si = util.build_typesafe_property(float)
+    """constant for plane wave laser without focus (unitless)"""
+    huygens_surface_positions = util.build_typesafe_property(typing.List[typing.List[int]])
+    """Position in cells of the Huygens surface relative to start/
+       edge(negative numbers) of the total domain"""
```

Comment:
here I  do not understand the class-level definition

Reply:
@PrometheusPi see comment above